apps/native-component-list/src/screens/DeviceScreen.tsx

```diff
@@ -0,0 +1,168 @@
+import React from 'react';
+import { ScrollView, View, Text, Button, Platform } from 'react-native';
+import * as Device from 'expo-device';
+import HeadingText from '../components/HeadingText';
+import MonoText from '../components/MonoText';
+
+interface DeviceConstantName {
+  name?: string;
+  value?: any;
+}
+
+interface State {
+  allFeatures?: string;
+  maxMemory?: number;
+  isSideLoadingEnabled?: string;
+  uptime?: number;
+  isRooted?: string;
+  deviceType?: Device.DeviceType;
+}
+
+class DeviceConstants extends React.Component<DeviceConstantName> {
+  render() {
+    let { name, value } = this.props;
+    if (typeof value === 'boolean') {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {String(value)}</MonoText>
+        </View>
+      );
+    } else {
+      return (
+        <View style={{ marginBottom: 10 }}>
+          <HeadingText>{name}</HeadingText>
+          <MonoText> {value}</MonoText>
+        </View>
+      );
+    }
+  }
+}
+
+export default class DeviceScreen extends React.Component<{}, State> {
+  static navigationOptions = {
+    title: 'Device',
+  };
+
+  state = {
+    allFeatures: '',
+    maxMemory: 0,
+    uptime: 0,
+    isSideLoadingEnabled: '',
+    isRooted: '',
+    deviceType: Device.DeviceType.UNKNOWN
+  };
+
+  _getAllFeatures = () => {
+    Device.getPlatformFeaturesAsync().then(allFeatures => {
+      let result = '';
+      allFeatures.forEach(feature => {
+        result += feature;
+        result += '\n';
+      });
+      this.setState({ allFeatures: result });
+    });
+  };
+  _getMaxMemory = () => {
+    Device.getMaxMemoryAsync().then(maxMemory => {
+      this.setState({ maxMemory: maxMemory });
+    });
+  };
+  _getUptime = () => {
+    Device.getUptimeAsync().then(uptime => {
+      this.setState({ uptime: uptime });
```

Comment:
```suggestion
      this.setState({ uptime });
```
I would also encourage you to configure your code editor so it displays warnings like 'use property shorthand' — they help follow conventions!

Reply:
Got it! Thank u!